### ISSUE TITLE
Dask labextension does not work in latest pangeo-notebook image

### ISSUE THREAD

**Issue by rabernat** [pangeo-data/pangeo-docker-images#177]
This is a duplicate of https://github.com/pangeo-gallery/default-binder/issues/21, but I'm positing it here for greater visibility.

In our latest pangeo-notebook images, pasting the dashboard link into the Dask labextension dialog has no effect. It does not recognize the link, and none of the buttons become active. I have tried this on several recent

![image](https://user-images.githubusercontent.com/1197350/103908821-f17f3f00-50d0-11eb-9243-d804f5cda489.png)

Here is a binder link for our classic example notebook using the latest pangeo-gallery/default-binder, which points to the latest pangeo-notebook docker image (c8c0d72).

https://binder.pangeo.io/v2/gh/pangeo-gallery/default-binder/64d772a/?urlpath=git-pull?repo=https://github.com/pangeo-gallery/physical-oceanography%26amp%3Burlpath=lab/tree/physical-oceanography/01_sea-surface-height.ipynb%3Fautodecode

All of our JupyterHubs are running `pangeo/pangeo-notebook:2020.12.08`. So something since then has broken the labextension.

**Comment by scottyhq** [comment]
do we know if the labextension worked previously (based on this [PR](https://github.com/pangeo-gallery/default-binder/commit/e59fea1f7ed260b7a61fedf4e98d5aa52f98f6f2#diff-b5e5dd909a2330cdff273c344a438825c22da1acc18a3a1880f5b86a7bda97c2) it looks like previous default binder pointed to 2020.12.16 in the default-binder repository) 

we don't programmatically test to make sure the labextension dashboard works with new images, it's always been up to people running notebooks and noticing problems like this. 

you can check version differences between images with URLs like this to figure out changed package versions https://github.com/pangeo-data/pangeo-docker-images/compare/2020.12.16..2021.01.07

**Comment by scottyhq** [comment]
i'm trying this out w/ 2021.01.07 and seeing errors like these in the browser JavaScript Console when i try to connect:

```
[Error] Failed to load resource: the server responded with a status of 404 () (controls.60728b5f435063d94b8f.js.map, line 0)
https://hub.aws-uswest2-binder.pangeo.io/user/scottyhq-pageodev-binder-image-1kr0x54c/static/lab/vendors~@jupyter-widgets/controls.60728b5f435063d94b8f.js.map
https://hub.aws-uswest2-binder.pangeo.io/user/scottyhq-pageodev-binder-image-1kr0x54c/static/lab/vendors~main.0d37c9e5f02fa65237ee.js.map
https://hub.aws-uswest2-binder.pangeo.io/user/scottyhq-pageodev-binder-image-1kr0x54c/static/lab/main.cfb9280d54fcb641c354.js.map
https://hub.aws-uswest2-binder.pangeo.io/user/scottyhq-pageodev-binder-image-1kr0x54c/individual-plots.json?1610307968106
```

Right now you can only access K8s pod logs as an admin. I see a lot of the following messages in my user pod (`kubectl logs -n prod    jupyter-scottyhq-2dpageodev-2dbinder-2dimage-2d1kr0x54c`)
```
[W 19:50:10.849 LabApp] https://hub.aws-uswest2-binder.pangeo.io/services/dask-gateway/clusters/prod.6f1c7b8cf1dd44be908888e893705ced/ does not seem to host a dask dashboard
[W 19:50:14.936 LabApp] https://hub.aws-uswest2-binder.pangeo.io/services/dask-gateway/clusters/prod.6f1c7b8cf1dd44be908888e893705ced/ does not seem to host a dask dashboard
```

However, opening the client dashboard link works just fine:
https://hub.aws-uswest2-binder.pangeo.io/services/dask-gateway/clusters/prod.6f1c7b8cf1dd44be908888e893705ced/status

maybe @TomAugspurger, @jacobtomlinson, @ian-r-rose, or @jsignell can help figure this out

**Comment by scottyhq** [comment]
finally, `kubectl logs -n prod traefik-prod-dask-gateway-5dd6c8dddf-tqzbn` shows: 
```
time="2021-01-09T20:10:29Z" level=error msg="Error while connection to backend: dial tcp 192.168.172.77:8786: connect: connection refused"
time="2021-01-09T20:10:29Z" level=error msg="the service \"prod-dask-f1ad5686e1084d8a864b5f45c1bf6bdf-81d549b3eca072ec7089@kubernetescrd\" does not exist" entryPointName=web routerName=prod-dask-f1ad5686e1084d8a864b5f45c1bf6bdf-81d549b3eca072ec7089@kubernetescrd
time="2021-01-09T20:10:30Z" level=error msg="subset not found for prod/dask-f1ad5686e1084d8a864b5f45c1bf6bdf" providerName=kubernetescrd ingress=dask-f1ad5686e1084d8a864b5f45c1bf6bdf namespace=prod
time="2021-01-09T20:10:30Z" level=error msg="Cannot create service: subset not found" providerName=kubernetescrd ingress=dask-f1ad5686e1084d8a864b5f45c1bf6bdf namespace=prod serviceName=dask-f1ad5686e1084d8a864b5f45c1bf6bdf servicePort=8786
time="2021-01-09T20:10:32Z" level=error msg="subset not found for prod/dask-f1ad5686e1084d8a864b5f45c1bf6bdf" ingress=dask-f1ad5686e1084d8a864b5f45c1bf6bdf namespace=prod providerName=kubernetescrd
time="2021-01-09T20:10:32Z" level=error msg="Cannot create service: subset not found" namespace=prod serviceName=dask-f1ad5686e1084d8a864b5f45c1bf6bdf providerName=kubernetescrd ingress=dask-f1ad5686e1084d8a864b5f45c1bf6bdf servicePort=8786
```

**Comment by rabernat** [comment]
Thanks so much for digging into this @scottyhq!

**Comment by TomAugspurger** [comment]
I think that the line https://github.com/dask/dask-labextension/blob/03de8be8df6d6fb851f6f0079332b315d2ba337e/dask_labextension/dashboardhandler.py#L34 is throwing an error. That seems to be there just to follow any redirects.

If we avoid an exception there with `raise_error=False` then I think the rest of it will go through. The next request at `/individual-plots.json` *does* work: https://github.com/dask/dask-labextension/blob/03de8be8df6d6fb851f6f0079332b315d2ba337e/dask_labextension/dashboardhandler.py#L38-L43

I'm not sure what changed so that requests to `https://<hub>/services/dask-gateway/clusters/<cluster>` fail but `https://<hub>/services/dask-gateway/clusters/<cluster>/status/` succeed, but I'll make a PR updating dask-labextension.

**Comment by consideRatio** [comment]
> [W 2021-01-10 20:18:48.353 SingleUserNotebookApp _version:70] jupyterhub version 1.2.2 != jupyterhub-singleuser version 1.3.0. This could cause failure to authenticate and result in redirect loops!

@scottyhq in this case, I think this is harmless since JupyterHub 1.3 / JupyterHub 1.2 doesn't contain changes that would cause issues afaik.

**Comment by scottyhq** [comment]
since it seems like it might be a while to sort out a fix, i've just pinned dask-labextension=3 in the pangeo-notebook metapackage (conda-forge/pangeo-notebook-feedstock#54). i'll tag a new docker image today with a functional labextension. we can manually test dask-labextension 4 or 5 on binder before re-incorporating.

**Comment by rabernat** [comment]
I just tested, and this is fixed by #54. 🎉 

We can close this but leave open the issue in dask-labextension.

Thanks so much @scottyhq for your help!
